ports/esp32/modnetwork.c

```diff
@@ -275,11 +275,52 @@ STATIC mp_obj_t esp_disconnect(mp_obj_t self_in) {
 
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
-STATIC mp_obj_t esp_status(mp_obj_t self_in) {
+
+STATIC mp_obj_t esp_status(size_t n_args, const mp_obj_t *args) {
+
+    //wlan_if_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+    wifi_mode_t mode;
+    wifi_sta_list_t station_list;
+    wifi_sta_info_t* stations;
+    char sta_mac[18];
+    mp_obj_t list = mp_obj_new_list(0,NULL);
+
+
+    if (n_args==1) {
+	    // no arguments -> return none
+	    return mp_const_none;
+    }
+    
+    switch ((uintptr_t)args[1]) {
+	// one argument -> return results in list
+	//mp_obj_t list = mp_obj_new_list(0,NULL);
+
+        case (uintptr_t)MP_OBJ_NEW_QSTR(MP_QSTR_stations) :
+	    // return number of connected stations, only if in soft-AP mode
+	    ESP_EXCEPTIONS(esp_wifi_get_mode(&mode));
+	    if ((mode & WIFI_MODE_AP)==0) {
+	        // we will only print stations in soft-AP mode, i.e. WIFI_MODE_AP
+	        mp_raise_ValueError("soft-AP mode must be activated to retrieve stations");
+	    } else {
+		ESP_EXCEPTIONS(esp_wifi_ap_get_sta_list(&station_list));
+		stations = (wifi_sta_info_t*)(station_list.sta);
+		for (int i=0; i<station_list.num; i++) {
+		    sprintf(sta_mac,"%02x:%02x:%02x:%02x:%02x:%02x",stations[i].mac[0],stations[i].mac[1],stations[i].mac[2],stations[i].mac[3],stations[i].mac[4],stations[i].mac[5]);
+                    mp_obj_t *s = mp_obj_new_str(sta_mac, 17);
```

Comment:
Please just return a bytes object with the raw 6-byte MAC address, similar to how `wlan.config('mac')` works.

Also, does the ESP IDF API allow to get the RSSI of the connected station?  If so that would be good to return as a tuple like `(MAC, RSSI)`.

Reply:
Nope, the API in its current version only returns the mac address. No RSSI.

Reply:
> Nope, the API in its current version only returns the mac address. No RSSI.

That's a shame.  I see that the ESP8266 doesn't support RSSI either for connected STAs.

So the options are:
1. make it return just a list of MAC addresses
2. make it return a 1-tuple with the element being the MAC address

The benefit of the latter is that it allows to add more members to the tuple later on (eg RSSI if it becomes available, and IP address of the client) and retain backwards compatibility (as long as you always use elem[0] to get the MAC address).

Some other ports may support more info in the returned list so it's good to think how to support them as well as ESP32, so the API is the same across all ports.

Reply:
How about returning using a named tuple. That would allow for easier extension in future if more information is to be returned in the tuple.